lib/cart.rb

```diff
@@ -0,0 +1,59 @@
+require_relative "menu_item"
+require_relative "menu"
+require_relative "order_item"
+require_relative "customer"
+
+class Cart
+  attr_accessor :cart
+  attr_accessor :total_cost
+
+  def initialize
+    @cart = []
+    @total_cost = 0
+  end
+
+  @@instance = Cart.new
```

Comment:
That is a good question and my understanding of the issues involved is incomplete, imperfect, and probably at least partially wrong.  But, as it stands now, understood that the answer to your question relates to the use of the singleton design pattern and the fact that we do not want more than one instance of this class running around, at least in the more complicated Takeaway program (mea culpa).